pdns/dnsdistdist/doh.cc

```diff
@@ -620,6 +620,9 @@ static void processDOHQuery(DOHUnitUniquePtr&& du)
       if (du->response.empty()) {
         du->response = std::move(du->query);
       }
+      auto dh = const_cast<struct dnsheader*>(reinterpret_cast<const struct dnsheader*>(du->response.data()));
+
+      handleResponseSent(qname, QType(qtype), 0., du->ids.origDest, ComboAddress(), du->response.size(), *dh, dnsdist::Protocol::DoH);
```

Comment:
The dnsdist CI failure is a real issue. In doh.cc, we can get `ProcessQueryResult::SendAnswer` with an answer that has been generated from Lua, for example, and such an answer does not have to be a valid DNS packet (it could be any kind of content that a HTTP client can handle, like HTML). This should be enough to handle almost all cases, and make sure that we do not read past our buffer in any case:
```suggestion
      if (du->response.size() > sizeof(dnsheader) && du->contentType.empty()) {
        auto dh = reinterpret_cast<const struct dnsheader*>(du->response.data());

        handleResponseSent(qname, QType(qtype), 0., du->ids.origDest, ComboAddress(), du->response.size(), *dh, dnsdist::Protocol::DoH);
      }
```